pkg/url/url.go

```diff
@@ -88,12 +90,17 @@ func Get(client *occlient.Client, localConfig *config.LocalConfigInfo, urlName s
 }
 
 // GetIngressOrRoute returns ingress/route spec for given URL name
-func GetIngressOrRoute(client *occlient.Client, kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string, routeSupported bool) (URL, error) {
+func GetIngressOrRoute(client *occlient.Client, kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, containerEndpointsMap map[string]map[string]parsercommon.Endpoint, componentName string, routeSupported bool) (URL, error) {
```

Comment:
instead of `containerEndpointsMap` I think we can use []common.DevfileComponent, that way in a best/average case  we won't need to loop over the entire component data to form the map.

Reply:
updated